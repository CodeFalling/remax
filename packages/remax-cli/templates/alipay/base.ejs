<template name="REMAX_TPL">
  <block a:for="{{root.children}}" a:key="{{item.id}}">
    <template is="REMAX_TPL_1" data="{{item}}"/>
  </block>
</template>

<template name="REMAX_TPL_1">
  <block>
    <block a:if="{{item && item.text}}">
      <!-- string -->
      {{item.text}}
    </block>

    <% for (let component of components) { %>
      <%- include('./component.ejs', {
        type: component.type,
        props: component.props,
        id: component.id,
      }) %>
    <% } %>
  </block>
</template>
