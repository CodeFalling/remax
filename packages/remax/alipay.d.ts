import 'mini-types';
export * from './lib/adapters/alipay';
