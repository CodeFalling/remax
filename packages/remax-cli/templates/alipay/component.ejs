<% camelCased = function(str) {
  if (str.indexOf('-') !== -1) {
    return str;
  }
  return str.replace(/-([a-z0-9])/g, function (g) { return g[1].toUpperCase();
})}%>

<template name="REMAX_TPL_1_<%=id%>">
  <<%=id%> <% for(let key of props) { %> <%=key%>="{{item.props['<%=camelCased(key)%>']}}" <% } %>>
    <% if (id === 'swiper') { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <swiper-item key="{{item.props.key}}">
        <block a:for="{{item.children}}" a:key="{{item.id}}">
          <template is="REMAX_TPL_1_CONTAINER" data="{{item: item}}" />
        </block>
      </swiper-item>
    </block>
    <% } else if (id === 'picker') { %>
    <view>
      <template is="REMAX_TPL_1_CONTAINER" data="{{item: item.children[0]}}" />
    </view>
    <% } else if (id === 'picker-view') { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <picker-view-column>
        <view a:for="{{item.children}}">
          <template is="REMAX_TPL_1_CONTAINER" data="{{item}}" />
        </view>
      </picker-view-column>
    </block>
    <% } else { %>
    <block a:for="{{item.children}}" a:key="{{item.id}}">
      <template is="REMAX_TPL_1_CONTAINER" data="{{item: item}}" />
    </block>
    <% } %>
  </<%=id%>>
</template>