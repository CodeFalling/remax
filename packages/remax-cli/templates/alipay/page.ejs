<import-sjs name="helper" from="<%= jsHelper %>"/>
<import src="<%= baseTemplate %>"/>
<template is="REMAX_TPL" data="{{tree: helper.reduce(action)}}"/>
