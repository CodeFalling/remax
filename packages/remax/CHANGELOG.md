# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/remaxjs/remax/compare/v1.0.18...v1.1.0) (2019-09-22)

### Bug Fixes

- 修正微信 Image mode 属性 type 定义 ([#240](https://github.com/remaxjs/remax/issues/240)) ([c1b5687](https://github.com/remaxjs/remax/commit/c1b5687))
- **wechat:** 添加微信遗漏的云开发 API ([#247](https://github.com/remaxjs/remax/issues/247)) ([4298813](https://github.com/remaxjs/remax/commit/4298813)), closes [#246](https://github.com/remaxjs/remax/issues/246)

# [1.1.0-alpha.5](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.4...v1.1.0-alpha.5) (2019-09-07)

# [1.1.0-alpha.4](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.3...v1.1.0-alpha.4) (2019-09-07)

# [1.1.0-alpha.3](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.2...v1.1.0-alpha.3) (2019-09-07)

### Bug Fixes

- 删除头条里没必要的 api ([f5af7aa](https://github.com/remaxjs/remax/commit/f5af7aa))
- 完善头条组件定义 ([aaf3a27](https://github.com/remaxjs/remax/commit/aaf3a27))

# [1.1.0-alpha.2](https://github.com/remaxjs/remax/compare/v1.0.15...v1.1.0-alpha.2) (2019-09-05)

### Features

- 新增 unstable_batchedUpdates ([5b485b4](https://github.com/remaxjs/remax/commit/5b485b4))

# [1.1.0-alpha.1](https://github.com/remaxjs/remax/compare/v1.0.11...v1.1.0-alpha.1) (2019-09-02)

### Features

- 支持 App 是一个 React 组件 ([9b5bc84](https://github.com/remaxjs/remax/commit/9b5bc84))

# [1.1.0-alpha.0](https://github.com/remaxjs/remax/compare/v1.0.6...v1.1.0-alpha.0) (2019-08-28)

### Features

- 支持字节跳动小程序 ([d7adadc](https://github.com/remaxjs/remax/commit/d7adadc))

## [1.0.18](https://github.com/remaxjs/remax/compare/v1.0.17...v1.0.18) (2019-09-18)

**Note:** Version bump only for package remax

## [1.0.17](https://github.com/remaxjs/remax/compare/v1.0.16...v1.0.17) (2019-09-18)

### Bug Fixes

- 修复无法使用 catch 事件和 dataset 的问题 ([44485b4](https://github.com/remaxjs/remax/commit/44485b4)), closes [#219](https://github.com/remaxjs/remax/issues/219) [#231](https://github.com/remaxjs/remax/issues/231)

# [1.1.0-alpha.5](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.4...v1.1.0-alpha.5) (2019-09-07)

**Note:** Version bump only for package remax

# [1.1.0-alpha.4](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.3...v1.1.0-alpha.4) (2019-09-07)

**Note:** Version bump only for package remax

# [1.1.0-alpha.3](https://github.com/remaxjs/remax/compare/v1.1.0-alpha.2...v1.1.0-alpha.3) (2019-09-07)

### Bug Fixes

- 删除头条里没必要的 api ([f5af7aa](https://github.com/remaxjs/remax/commit/f5af7aa))
- 完善头条组件定义 ([aaf3a27](https://github.com/remaxjs/remax/commit/aaf3a27))

# [1.1.0-alpha.2](https://github.com/remaxjs/remax/compare/v1.0.15...v1.1.0-alpha.2) (2019-09-05)

### Bug Fixes

- 修正支付宝 api 类型定义 ([844624a](https://github.com/remaxjs/remax/commit/844624a))

### Features

- 新增 unstable_batchedUpdates ([5b485b4](https://github.com/remaxjs/remax/commit/5b485b4))

# [1.1.0-alpha.1](https://github.com/remaxjs/remax/compare/v1.0.11...v1.1.0-alpha.1) (2019-09-02)

### Features

- 支持 App 是一个 React 组件 ([9b5bc84](https://github.com/remaxjs/remax/commit/9b5bc84))

# [1.1.0-alpha.0](https://github.com/remaxjs/remax/compare/v1.0.6...v1.1.0-alpha.0) (2019-08-28)

### Features

- 支持字节跳动小程序 ([d7adadc](https://github.com/remaxjs/remax/commit/d7adadc))

# [1.1.0-alpha.1](https://github.com/remaxjs/remax/compare/v1.0.11...v1.1.0-alpha.1) (2019-09-02)

### Features

- 支持 App 是一个 React 组件 ([9b5bc84](https://github.com/remaxjs/remax/commit/9b5bc84))

# [1.1.0-alpha.0](https://github.com/remaxjs/remax/compare/v1.0.6...v1.1.0-alpha.0) (2019-08-28)

### Bug Fixes

- 修正支付宝组件的类型定义 ([#154](https://github.com/remaxjs/remax/issues/154)) ([9050ca1](https://github.com/remaxjs/remax/commit/9050ca1))

### Features

- 支持字节跳动小程序 ([d7adadc](https://github.com/remaxjs/remax/commit/d7adadc))

## [1.0.16](https://github.com/remaxjs/remax/compare/v1.0.15...v1.0.16) (2019-09-16)

### Bug Fixes

- 修复 onReady 回调 ([f6e276d](https://github.com/remaxjs/remax/commit/f6e276d)), closes [#203](https://github.com/remaxjs/remax/issues/203)
- 修正支付宝 api 类型定义 ([844624a](https://github.com/remaxjs/remax/commit/844624a))
- **wechat:** 更正 Image 和 Button 的 type 定义 ([7ba51d3](https://github.com/remaxjs/remax/commit/7ba51d3))

## [1.0.15](https://github.com/remaxjs/remax/compare/v1.0.14...v1.0.15) (2019-09-04)

**Note:** Version bump only for package remax

## [1.0.14](https://github.com/remaxjs/remax/compare/v1.0.13...v1.0.14) (2019-09-03)

### Bug Fixes

- 修复支付宝 Button 组件缺少的 style 属性 ([9e52a2d](https://github.com/remaxjs/remax/commit/9e52a2d))
- 修正微信组件及 API 类型定义 ([4630ff4](https://github.com/remaxjs/remax/commit/4630ff4))

## [1.0.13](https://github.com/remaxjs/remax/compare/v1.0.12...v1.0.13) (2019-09-02)

### Bug Fixes

- 修复生命周期 hooks 里的闭包变量不更新的问题 ([3b5e039](https://github.com/remaxjs/remax/commit/3b5e039))

## [1.0.12](https://github.com/remaxjs/remax/compare/v1.0.11...v1.0.12) (2019-09-02)

**Note:** Version bump only for package remax

## [1.0.11](https://github.com/remaxjs/remax/compare/v1.0.10...v1.0.11) (2019-08-31)

### Bug Fixes

- 修复微信 api 的类型定义 ([49e8d34](https://github.com/remaxjs/remax/commit/49e8d34))
- 完善微信组件类型定义 ([#161](https://github.com/remaxjs/remax/issues/161)) ([356fcbf](https://github.com/remaxjs/remax/commit/356fcbf))

## [1.0.10](https://github.com/remaxjs/remax/compare/v1.0.9...v1.0.10) (2019-08-30)

### Bug Fixes

- 修复支付宝基础组件类型 ([1b30b49](https://github.com/remaxjs/remax/commit/1b30b49))

## [1.0.9](https://github.com/remaxjs/remax/compare/v1.0.8...v1.0.9) (2019-08-29)

**Note:** Version bump only for package remax

## [1.0.8](https://github.com/remaxjs/remax/compare/v1.0.7...v1.0.8) (2019-08-28)

**Note:** Version bump only for package remax

## [1.0.7](https://github.com/remaxjs/remax/compare/v1.0.6...v1.0.7) (2019-08-28)

### Bug Fixes

- 修正支付宝 events 事件不触发的问题 ([dbf6e09](https://github.com/remaxjs/remax/commit/dbf6e09))
- 修正支付宝组件的类型定义 ([#154](https://github.com/remaxjs/remax/issues/154)) ([9050ca1](https://github.com/remaxjs/remax/commit/9050ca1))

## [1.0.6](https://github.com/remaxjs/remax/compare/v1.0.5...v1.0.6) (2019-08-28)

### Bug Fixes

- 修复 useShareAppMessage 的回调会被多次调用的问题 ([20adb1f](https://github.com/remaxjs/remax/commit/20adb1f)), closes [#172](https://github.com/remaxjs/remax/issues/172)

## [1.0.5](https://github.com/remaxjs/remax/compare/v1.0.4...v1.0.5) (2019-08-27)

### Bug Fixes

- 修复 onShareAppMessage 拿不到参数的问题 ([9189f01](https://github.com/remaxjs/remax/commit/9189f01)), closes [#169](https://github.com/remaxjs/remax/issues/169)
- 微信 Swiper 自动播放不会受 state 变化影响 ([bdc9c67](https://github.com/remaxjs/remax/commit/bdc9c67))

### Performance Improvements

- 提升模板渲染性能 ([9af03f9](https://github.com/remaxjs/remax/commit/9af03f9))

## [1.0.4](https://github.com/remaxjs/remax/compare/v1.0.3...v1.0.4) (2019-08-24)

**Note:** Version bump only for package remax

## [1.0.3](https://github.com/remaxjs/remax/compare/v1.0.2...v1.0.3) (2019-08-24)

### Bug Fixes

- 修复支付宝的 PickerView 组件 ([731b28e](https://github.com/remaxjs/remax/commit/731b28e))

## [1.0.2](https://github.com/remaxjs/remax/compare/v1.0.1...v1.0.2) (2019-08-23)

### Bug Fixes

- style 为 null/undefined 时报错的问题 ([28f4367](https://github.com/remaxjs/remax/commit/28f4367))
- 生命周期 hooks 不能在组件中使用的问题 ([4fa5695](https://github.com/remaxjs/remax/commit/4fa5695))
