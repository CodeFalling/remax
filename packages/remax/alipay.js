export * from './esm/adapters/alipay';
