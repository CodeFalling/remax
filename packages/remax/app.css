*,
*:after,
*:before {
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

html {
  line-height: 1.2;
}

body {
  user-select: none;
  font-size: 16px;
  font-family: PingFang SC, Helvetica Neue;
  background: transparent;
}

*[contenteditable] {
  -webkit-user-select: auto !important;
}

*:focus {
  outline: none;
}

a {
  background: transparent;
  text-decoration: none;
  outline: none;
}

.remax-page.with-tab-bar {
  padding-bottom: 96px;
}

.remax-page {
  min-height: 100vh;
  box-sizing: border-box;
}

.remax-page.disable-scroll {
  overflow: hidden;
}

.rmc-pull-to-refresh-transition {
  transition: transform 0.3s;
}

.rmc-pull-to-refresh-wrapper {
  overflow: hidden;
}

.rmc-pull-to-refresh-indicator {
  color: grey;
  text-align: center;
  height: 100px;
}

.rmc-pull-to-refresh-indicator {
  margin-top: -100px;
}

.remax-pull-to-refresh-loading {
  width: 100px;
  height: 100px;
}

.remax-tab-bar {
  height: 96px;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: #fff;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  z-index: 999999;
}

.remax-tab-bar .remax-tab-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.remax-tab-bar .remax-tab-item .remax-tab-item-image {
  height: 34px;
  width: 34px;
  background-size: contain;
  background-position: 50%;
  background-repeat: no-repeat;
}

.remax-tab-bar .remax-tab-item .remax-tab-item-title {
  font-size: 12px;
  color: #333;
  transform: scale(0.7);
}

.remax-tab-bar .remax-tab-item .remax-tab-item-title.active {
  color: #108ee9;
}
