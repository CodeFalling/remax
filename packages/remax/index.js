export {
  render,
  createAppConfig,
  createPageConfig,
  PageProps,
  usePageInstance,
  useQuery,
  unstable_batchedUpdates,
} from '@remax/runtime';
