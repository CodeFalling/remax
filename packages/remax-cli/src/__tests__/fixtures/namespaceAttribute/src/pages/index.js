import * as React from 'react';
import { View } from 'remax/alipay';
import NativeComponent from '@/components/native-component';
import './index.css';

export default function Index() {
  const props = {};
  return (
    <>
      <View ns:prop="ns:value" />
      <NativeComponent ns:prop="ns:value" {...props} />
    </>
  );
}
