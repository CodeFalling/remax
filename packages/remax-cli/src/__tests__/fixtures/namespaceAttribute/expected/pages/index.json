{
  "usingComponents": {
    "native-component-0": "../components/native-component/index"
  }
}